### PR TITLE
Docs improvement

### PR DESCRIPTION
**Pull request by majk-p**
This PR improves the `Sender` and `Consumer` sections of the documentation. It also adds some new sections: `localstack`, `xml` and `json`. The improvements include some examples, explanation of yet uncovered areas and typo fixes

### CODE REVIEW

File: docs/core-concepts/Consumer.md [hunk 1]
```diff
@@ -30,3 +30,67 @@ The end user usually doesn't instantiate the `Consumer` directly. Instead they w
...
 
 This abstraction comes with a lot of useful manipulators as well as `Semigroup`, `Monoid`, `Functor` and `Monad` instances. Please refer to [`Consumer.scala` sources](https://github.com/ocadotechnology/pass4s/blob/main/kernel/src/main/scala/com/ocadotechnology/pass4s/kernel/Consumer.scala) and the scaladocs.
+
+# Basic usage
+
+Here's a simple consumer implementation configured to use with our [localstack setup](localstack). If you want to run it locally, simply save the file somewhere and run it using [scala-cli](https://scala-cli.virtuslab.org/install) using `scala-cli run filename.scala`
+
+```scala
+//> using scala "2.13"
+//> using lib "com.ocadotechnology::pass4s-kernel:0.2.2"
```
Comment (jsienniak) [inline docs/core-concepts/Consumer.md:40]: 0.3.1?

Comment (jsienniak) [reply docs/core-concepts/Consumer.md:40]: here and in other places

File: docs/core-concepts/Consumer.md [hunk 2]
```diff
@@ -30,3 +30,67 @@ The end user usually doesn't instantiate the `Consumer` directly. Instead they w
...
+          .consume(message => IO.println(s"Received message: $message")) // bind consumer logic
+          .background // run in background
+          .void
+          .use(_ => IO.never)
+    }
+  }
+}
+```
+
+This is a rather raw way of using consumer, you might want to use [MessageProcessor](modules/message-processor) for more elasticity and enrichment syntax.
```
Comment (jsienniak) [inline docs/core-concepts/Consumer.md:96]:
```suggestion
This is a rather raw way of using consumer, you might want to use [MessageProcessor](modules/message-processor) for more elasticity and enriched syntax.
```

File: docs/localstack.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+## Resources
+
+This setup comes with batteries included, meaning it not only does set up the service, but also some basic resources for you to play with. The provided resources are:
+
+ - Topic `local_sns` to write your messages onto
+ - Queue `local_queue` to read messages from
+ - Subscription between the two, making the messages from `local_sns` to be pushed to `local_queue`
+ - Bucket `large-messages` in case you want to use [s3 proxy](modules/s3proxy)
+
+Please notice how for queue creation we use `--attributes 'RawMessageDelivery=true'`. This is done intentionally, make sure to use this attribute with your production setup to avoid communication issues.
```
Comment (jsienniak) [inline docs/localstack.md:69]: would be good to explain why this is important (link to aws docs maybe?)

**Review by jsienniak — APPROVED** [review]

File: docs/core-concepts/Consumer.md (hunk 1 above)
Comment (majk-p) [reply docs/core-concepts/Consumer.md]: good catch, fixed

File: docs/localstack.md (hunk 3 above)
Comment (majk-p) [reply docs/localstack.md]: Added some more explanation + a link to AWS docs

**Pull request merged**